pkg/codegen/java/gen_program.go

```diff
@@ -568,12 +572,15 @@ func makeResourceName(baseName string, suffix string) string {
 }
 
 func (g *generator) findFunctionSchema(function string) (*schema.Function, bool) {
-	for _, pkg := range g.program.Packages() {
-		if pkg.Functions != nil {
-			for _, functionSchema := range pkg.Functions {
-				if functionSchema != nil && strings.HasSuffix(functionSchema.Token, names.LowerCamelCase(function)) {
-					return functionSchema, true
+	function = names.LowerCamelCase(function)
+	for _, pkg := range g.program.PackageReferences() {
+		for it := pkg.Functions().Range(); it.Next(); {
+			if strings.HasSuffix(it.Token(), function) {
```

Comment:
I believe because `it.Token()` contains the `pkg:module:functionName` so here the check determines whether `function` is `functionName` (from the token)